core/src/main/java/kafka/server/share/SharePartition.java

```diff
@@ -456,6 +456,17 @@ public CompletableFuture<Void> maybeInitialize() {
                         stateBatch.lastOffset(), RecordState.forId(stateBatch.deliveryState()), stateBatch.deliveryCount(), null);
                     cachedState.put(stateBatch.firstOffset(), inFlightBatch);
                 }
+
+                long initialOffset = startOffset;
+                for (InFlightBatch currentStateBatch : cachedState.values()) {
+                    if (initialOffset != currentStateBatch.firstOffset()) {
+                        InFlightBatch newBatch = new InFlightBatch(EMPTY_MEMBER_ID, initialOffset, currentStateBatch.firstOffset() - 1,
+                            RecordState.AVAILABLE, (short) 0, null);
+                        cachedState.put(initialOffset, newBatch);
+                    }
+                    initialOffset = currentStateBatch.lastOffset() + 1;
+                }
```

Comment:
Can you move the processing to a method please.

Reply:
nit: may be you want to execute it inside the if condition just below `if (!cachedState.isEmpty()) {`